mlflow/azure/client.py

```diff
@@ -71,6 +71,7 @@ def patch_adls_file_upload(sas_url, local_file, start_byte, size, position, head
         "patch", request_url, data=data, headers=request_headers
     ) as response:
         rest_utils.augmented_raise_for_status(response)
+        response.close()
```

Comment:
We're using a context manager here. `response` is an instance of `requests.Resopnse` which should release the connection on `__exit__`.

Reply:
We would need this on `put` as well as `patch` - we've seen the hang once there too.